File: python-reprojector/pyreproj-py2many/tests/reprojector/test_get_proj_from_parameter.jl
module test_get_proj_from_parameter
using FromFile: @from
using PyCall
using Test
pyproj = pyimport("pyproj")

@from "../../__init__.jl" using pyreproj: InvalidFormatError
@from "../../__init__.jl" using pyreproj: Reprojector, _get_proj_from_parameter

function test_get_proj_from_parameter_()
    param_args = [
        "epsg:4326",
        "EPSG:4326",
        4326,
        "+proj=longlat +ellps=WGS84 +datum=WGS84 +no_defs",
        pyproj.Proj(init = "epsg:4326"),
    ]
    for param in param_args
        rp = Reprojector()
        proj = _get_proj_from_parameter(rp, param)
        @assert(pybuiltin(:isinstance)(proj, pyproj.Proj))
    end
end

function test_get_proj_from_parameter_invalid()
    param_args = [nothing, "invalid_parameter"]
    for param in param_args
        @test_throws InvalidFormatError begin
            rp = Reprojector()
            _get_proj_from_parameter(rp, param)
        end
    end
end

if abspath(PROGRAM_FILE) == @__FILE__
    test_get_proj_from_parameter_()
    test_get_proj_from_parameter_invalid()
end
end

File: libraries/python-reprojector/pyreproj/tests/reprojector/test_get_proj_from_parameter.py
# -*- coding: utf-8 -*-
import pytest

from pyreproj import InvalidFormatError
from pyreproj import Reprojector
from pyproj import Proj

@pytest.mark.parametrize("param", [
    "epsg:4326",
    "EPSG:4326",
    4326,
    "+proj=longlat +ellps=WGS84 +datum=WGS84 +no_defs",
    Proj(init='epsg:4326')
])
def test_get_proj_from_parameter(param):
    rp = Reprojector()
    proj = rp._get_proj_from_parameter(param)
    assert isinstance(proj, Proj)


@pytest.mark.parametrize("param", [
    None,
    "invalid_parameter"
])
def test_get_proj_from_parameter_invalid(param):
    with pytest.raises(InvalidFormatError):
        rp = Reprojector()
        rp._get_proj_from_parameter(param)
